### DISCUSSION TITLE
Optional binding in views?

### DISCUSSION THREAD

**Discussion by agruenberg** [Budibase/budibase#8177] [Help] [answered]
Is there any possibility to make an optional binding in views?
I would like to do something like this:

`select invoice_year, sum(amount) as cost from costs
{{#if cat }} where cost_category_id = {{ cat }}{{/if}}
group by invoice_year;`

where `cat` is the binding parameter.

**Comment by melohagan** [comment, marked as answer]
Hey @agruenberg 

Have you tried using a CASE statement? https://www.sqlshack.com/case-statement-in-sql/

**Reply by agruenberg** [reply]
Thank you very much. But what I am trying is to avoid the complete where clause conditionaly. I dont see how i could achieve this with a case statement.

**Reply by melohagan** [reply]
Well you may not be able to remove the where clause entirely, but you could have an arbitrary case that returns all of the rows, e.g. `where cost_category_id > 0`

This StackOverflow post might help: https://stackoverflow.com/questions/18629132/conditional-where-clause-in-sql-server

**Reply by agruenberg** [reply]
Thank You very much
